offix/src/main/java/org/aerogear/offix/Offix.kt

```diff
@@ -64,8 +78,28 @@ fun ApolloClient.enqueue(
             Log.d("Extension Callback * ", result)
 
             /*
-             Set the response received from the server in the onSucceess() of responseCallback.
+             1. Check if the response data is null or not. If it's null that means the conflict has happened.
+             2. Parse the json error response from the server.
+             3. Get the serverstate and the clientstate keys from the error response.
+             4. Set them in responseCallback's onConflictDetected() method.
+             5. The user resolves conflicts and send the mutation back to the library.
+             6. Make a recursive call to the enqueue() function and retry mutation again.
              */
+            if (response.data() == null && detectedConflict.isNotEmpty()) {
+                val serverClientData = getServerClientData(detectedConflict)
+
+                val retryMutation =
+                    responseCallback.onConflictDetected(serverClientData.serverData, serverClientData.clientData)
+
+                /* Make a recursive call to the enqueue method to retry the mutation
+                 */
+                Offline.apClient?.mutate(retryMutation)?.enqueue(this)
```

Comment:
Here, the code should be like this : 

```
retryMutation?.let{
    Offline.apClient?.mutate(it)?.enqueue(this)
}
```

Why, see my comment in the interface below.